### PR TITLE
Adds faraday with net-http-persistent adapter to uktt gem

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-882

### What?

I have added/removed/altered:

- [x] Added faraday to the uktt gem
- [x] Updated tests to reflect change in adapter

### Why?

I am doing this because:

- This is desired as part of debugging why connections can be a bit buggy on high loads in production

### AC

- [x] Supports retriable functionality
- [x] Supports redirects
- [x] Supports more efficient adapter
- [x] Does not change api of gem
- [x] Tested in private routes mode

### CODE REVIEW

File: lib/uktt/http.rb [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+require 'faraday_middleware'
 
 module Uktt
   class Http
     DEFAULT_BACKEND_SERVICE = 'uk'.freeze
     DEFAULT_PARSED_FORMAT = 'jsonapi'.freeze
-    DEFAULT_RETRIABLE_INTERVALS = [0.5, 1.0, 2.0, 2.5, 20.0].freeze
+    DEFAULT_RETRIABLE_OPTIONS = {
+      max: 2,
+      interval: 0.05,
```
Comment (jebw) [inline lib/uktt/http.rb]: This seems very low - if we're needing to retry, we could give the server chance to settle - eg 2 secs?

File: lib/uktt/http.rb [hunk 2]
```diff
@@ -16,26 +20,23 @@ def initialize(connection, options)
...
 
-    def self.build(host, version, format, public_routes, retriable_intervals = nil)
-      uri = URI(host)
-      connection = Net::HTTP.new(uri.host, uri.port)
-      connection.use_ssl = uri.scheme.include?('https')
+    def self.build(host, version, format, public_routes, retriable_options = nil)
+      connection = Faraday.new(url: host) do |faraday|
+        faraday.use FaradayMiddleware::FollowRedirects
+        faraday.adapter :net_http_persistent
+        faraday.response :logger if ENV['DEBUG_REQUESTS']
```
Comment (jebw) [inline lib/uktt/http.rb:33]: Is it worth sticking the `RaiseError` middleware in there?

**Review by jebw — COMMENTED** [review]
Looks good - wondering whether `RaiseError` would be useful - makes it easy to just handle the different status codes further up in the app

Also the 50ms, 100ms, 200ms seem too low for a retry sequence?  Much lower then they were previously?

File: lib/uktt/http.rb (hunk 1 above)
Comment (willfish) [reply lib/uktt/http.rb]: Yeah, you're right, here. I actually just copy and pasted the defaults in an example and forgot to review :)

Comment (willfish) [reply lib/uktt/http.rb]: I agree 2 seconds makes sense if we get some failure

File: lib/uktt/http.rb (hunk 2 above)
Comment (willfish) [reply lib/uktt/http.rb:33]: I've actually not used it before. I will have a gander

Comment (willfish) [reply lib/uktt/http.rb:33]: Added

File: lib/uktt/http.rb (hunk 1 above)
Comment (willfish) [reply lib/uktt/http.rb]: Done

**Review by jebw — APPROVED** [review]
LGTM - nice should hopefully help with the occasional flaky network connection

**Pull request merged**